lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py

```diff
@@ -339,6 +388,109 @@ def _generate_report(self, generated_assembly_ref_list, params):
 
         return report_output
 
+    def _generate_report_message(self, new_binned_contig_ref):
+        """
+        _generate_report_message: generate a report message for BinnedContig object
+        """
+
+        report_message = ''
+
+        binned_contig = self.dfu.get_objects({'object_refs': [new_binned_contig_ref]})['data'][0]
+        binned_contig_info = binned_contig.get('info')
+        binned_contig_name = binned_contig_info[1]
+        report_message += 'Generated BinnedContig: {} [reference {}]\n'.format(
+                                                                        binned_contig_name,
+                                                                        new_binned_contig_ref)
+
+        binned_contig_count = 0
+        total_bins = binned_contig.get('data').get('bins')
+        total_bins_count = len(total_bins)
+        bin_ids = []
+        for bin in total_bins:
+            binned_contig_count += len(bin.get('contigs'))
+            bin_ids.append(bin.get('bid'))
+
+        report_message += '--------------------------\nSummary:\n\n'
+        report_message += 'Binned contigs: {}\n'.format(binned_contig_count)
+        report_message += 'Total size of bins: {}\n'.format(total_bins_count)
+        report_message += 'Bin IDs:\n{}\n'.format('\n'.join(bin_ids))
+
+        return report_message
+
+    def _merge_bins(self, new_bin_id, bin_objects_to_merge):
+        """
+        _merge_bins: merge a list of bins into new_bin_id
+
+        """
+        total_contigs = {}
+        total_gc_count = 0
+        total_sum_contig_len = 0
+        total_cov_len = 0
+
+        for bin in bin_objects_to_merge:
+            total_contigs.update(bin.get('contigs'))
+            sum_contig_len = bin.get('sum_contig_len')
+            total_sum_contig_len += sum_contig_len
+            total_gc_count += round(sum_contig_len * bin.get('gc'), 5)
+            total_cov_len += round(sum_contig_len * bin.get('cov'), 5)
+
+        contig_bin = {
+            'bid': new_bin_id,
+            'contigs': total_contigs,
+            'n_contigs': len(total_contigs),
+            'gc': round(float(total_gc_count)/total_sum_contig_len, 5),
+            'sum_contig_len': total_sum_contig_len,
+            'cov': round(float(total_cov_len)/total_sum_contig_len, 5)
```

Comment:
The math (computing the weighted average for cov and GC) looks correct to me!  One minor point is that we probably shouldn't round the calculations on 434 and 435- just compute to full precision there and round at the end when we're saving the final value.